protocols/floodsub/src/layer.rs

```diff
@@ -359,8 +360,12 @@ impl NetworkBehaviour for Floodsub {
     ) {
         // We ignore successful sends or timeouts.
         let event = match event {
-            InnerMessage::Rx(event) => event,
-            InnerMessage::Sent => return,
+            Ok(InnerMessage::Rx(event)) => event,
+            Ok(InnerMessage::Sent) => return,
+            Err(e) => {
+                log::debug!("Failed to send floodsub message: {e}");
```

Comment:
Whilst we could do a fancier error handling here now, I don't think improving `libp2p-floodsub` has a high priority. Logging the error is also what we did before.